spring-xd-analytics/src/main/java/org/springframework/xd/analytics/metrics/redis/RedisFieldValueCounterRepository.java

```diff
@@ -54,20 +54,16 @@ public RedisFieldValueCounterRepository(RedisConnectionFactory connectionFactory
 		redisTemplate.afterPropertiesSet();
 	}
 
+	/*
+	 * Note: Handler implementations typically use increment() variants to save state. The save() contract
+	 * is to store the counter as a whole. Simplest approach is erase/rewrite.
+	 */
 	@Override
 	public <S extends FieldValueCounter> S save(S fieldValueCounter) {
-		String counterKey = getMetricKey(fieldValueCounter.getName());
-		if (this.redisTemplate.opsForValue().get(counterKey) == null) {
-			if (fieldValueCounter.getFieldValueCount().size() > 0) {
-				for (Map.Entry<String, Double> entry : fieldValueCounter.getFieldValueCount().entrySet()) {
-					increment(fieldValueCounter.getName(), entry.getKey(), entry.getValue());
-				}
-			}
-			else {
-				increment(fieldValueCounter.getName(), MARKER, 0);
-			}
+		delete(fieldValueCounter.getName());
+		for (Map.Entry<String, Double> entry : fieldValueCounter.getFieldValueCount().entrySet()) {
+			increment(fieldValueCounter.getName(), entry.getKey(), entry.getValue());
 		}
```

Comment:
The test in `RedisFieldValueCounterRepositoryTests` fails now.

----
	public void testCrud() {
		FieldValueCounter counter = new FieldValueCounter(tickersFieldValueCounterName);

		// Test with initial save
		FieldValueCounter fvTickersCounter = fieldValueCounterRepository.save(counter);
		assertThat(fieldValueCounterRepository.findOne(fvTickersCounter.getName()), notNullValue());
----
I think the previous code  was trying to fake saving a 0 value entry in redis to preserve 'i saved an empty FVC' semantics.  This seems to still be needed, I updated the save method to.

----
	public <S extends FieldValueCounter> S save(S fieldValueCounter) {
		delete(fieldValueCounter.getName());
		increment(fieldValueCounter.getName(), MARKER, 0);
		for (Map.Entry<String, Double> entry : fieldValueCounter.getFieldValueCount().entrySet()) {
			increment(fieldValueCounter.getName(), entry.getKey(), entry.getValue());
		}
		return fieldValueCounter;
	}
----

and it saves correctly now.  Will add as part of merge.